.github/workflows/release.yml

```diff
@@ -0,0 +1,114 @@
+name: Release
+on:
+  push:
+    tags:
+      - v*
+
+env:
+  ELIXIR_OPTS: "--warnings-as-errors"
+  ERLC_OPTS: "warnings_as_errors"
+  LANG: C.UTF-8
+
+jobs:
+  build_zips:
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - otp: 22
+            otp_version: 22.3
+          - otp: 23
+            otp_version: 23.3
+          - otp: 24
+            otp_version: 24.1
+            build_docs: build_docs
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 50
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp_version }}
+          version-type: strict
+      - name: Build Elixir Release
+        run: |
+          vsn=$(cat VERSION)
+          make Precompiled.zip
+          mv Precompiled-v${vsn}.zip elixir-otp-${{ matrix.otp }}.zip
+          echo "$PWD/bin" >> $GITHUB_PATH
+      - uses: actions/upload-artifact@v2
+        with:
+          name: elixir-otp-${{ matrix.otp }}.zip
+          path: elixir-otp-${{ matrix.otp }}.zip
+      - uses: actions/checkout@v2
+        with:
+          repository: elixir-lang/ex_doc
+          path: ex_doc
+        if: ${{ matrix.build_docs }}
+      - name: Build ex_doc
+        if: ${{ matrix.build_docs }}
+        run: |
+          mv ex_doc ../ex_doc
+          cd ../ex_doc
+          ../elixir/bin/mix do local.rebar --force, local.hex --force, deps.get, compile
+          cd ../elixir
+      - name: Build Docs
+        if: ${{ matrix.build_docs }}
+        run: |
+          vsn=$(cat VERSION)
+          make Docs.zip
+          mv Docs-v${vsn}.zip Docs.zip
+      - uses: actions/upload-artifact@v2
+        with:
+          name: Docs.zip
+          path: Docs.zip
+        if: ${{ matrix.build_docs }}
+  create_draft_release:
+    needs: build_zips
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 50
+      - uses: actions/download-artifact@v2
+        with:
+          name: elixir-otp-22.zip
+      - uses: actions/download-artifact@v2
+        with:
+          name: elixir-otp-23.zip
+      - uses: actions/download-artifact@v2
+        with:
+          name: elixir-otp-24.zip
```

Comment:
Can we download those from the matrix? This way when adding/removing OTP versions we only need to change one place in the file. :)

Reply:
To be clear, what I mean is that it would be nice if the second part of the snippet downloaded the files and generated the checksums based on the OTP versions of the matrix. :)

If it not possible, consider doing `shasum -a 256 *.zip`, so there is less repetition. Potentially downloading them to a common subdirectory to avoid unwarranted `.zip` files from showing up?

Reply:
Not sure if upload artifact can upload to the same destination. Let me play around with this a little bit. :)